package/MDAnalysis/core/universe.py

```diff
@@ -985,6 +986,208 @@ def add_Segment(self, **attrs):
         self.segments = SegmentGroup(np.arange(self._topology.n_segments), self)
         # return the new segment
         return self.segments[segidx]
+    
+    def add_TopologyObject(self, value, type=None, types=None, guessed=False,
```

Comment:
I know they're called `TopologyObject`s, but I really don't like this method name.  I'd rather have just `add_Bonds` `add_Angles` and `add_Dihedrals`, I think they're clearer (and more likely to be found by someone).  Also I'd not bother with the singular version and just force people to give a 1 length list if they want to add a single Bond

Reply:
Changed -- this also simplifies the type checking and tests. I'm still not sure how to document the 'order' and 'types' keywords, or whether to even leave them in for non-Bonds (especially order).